src/utils/parser.js

```diff
@@ -672,6 +672,24 @@ parser.get_orderby_clause = function f(queryObject) {
   return orderKeys.length ? util.format('ORDER BY %s', orderKeys.join(', ')) : ' ';
 };
 
+parser.get_groupby_clause = function f(queryObject) {
+  let groupbyKeys = [];
+
+  Object.keys(queryObject).forEach((k) => {
+    const queryItem = queryObject[k];
+
+    if (k.toLowerCase() === '$groupby') {
+      if (!(queryItem instanceof Array)) {
+        throw (buildError('model.find.invalidgroup'));
+      }
+
+      groupbyKeys = groupbyKeys.concat(queryItem);
+    }
+  });
+
+  return groupbyKeys.length ? util.format('GROUP BY %s', groupbyKeys.join(', ')) : ' ';
```

Comment:
I guess this will fail for keys with uppercased characters in them. For example field names like `project_id` will work fine, but `projectId` won't. using quotes like `GROUP BY "%s"` should solve the problem.

Also please add a test case for this type of scenarios.